automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -193,6 +196,106 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
           }(ownerAuthToken)
         }
       }
+
+      val testAttributes: Map[String, String] = Map("A-key" -> "A-value", "B-key" -> "B-value", "C-key" -> "C-value")
+      val attributeMap: AttributeMap = testAttributes.map(keyValuePairs => AttributeName("default", keyValuePairs._1) -> AttributeString(keyValuePairs._2))
+
+      "to add workspace attributes" in {
+        implicit val token: AuthToken = ownerAuthToken
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("add-attributes")) { workspaceName =>
+            val attributeUpdates = attributeMap.map(attrTuple => AddUpdateAttribute(attrTuple._1, attrTuple._2)).toList
+
+            Rawls.workspaces.updateAttributes(projectName, workspaceName, attributeUpdates)
+            eventually {
+              workspaceResponse(Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName)).workspace.attributes should be (attributeMap)
+            }
+          }
+        }
+      }
+
+      "to delete workspace attributes" - {
+        "from the top" in {
```

Comment:
I believe deleting ws attributes from the "top" "middle" and "bottom" of the table is testing the _ui table_ to make sure that it updates properly. For the rawls API tests, we just need to test a simple deletion, so top, middle and bottom don't matter. So, I would reduce the "delete workspace attributes" to just one test.

Reply:
Okay great, that will make the tests much simpler